src/webui/api/appcontroller.cpp

```diff
@@ -1083,6 +1084,15 @@ void AppController::defaultSavePathAction()
     setResult(BitTorrent::Session::instance()->savePath().toString());
 }
 
+void AppController::getDirectoriesAction()
+{
+    requireParams({u"path"_s});
+
+    const QString path = params().value(u"path"_s);
```

Comment:
You should at least reject empty Path and paths starting with `:`.

Reply:
Only absolute paths should be accepted.

Reply:
@Chocobo1 Why some paths could start with `:`?

Reply:
> @Chocobo1 Why some paths could start with `:`?

[The Qt Resource System](https://doc.qt.io/qt-6/resources.html)

Reply:
Oh, I forgot about this mechanism

Reply:
I added this check.